virttest/qemu_monitor.py

```diff
@@ -61,6 +61,18 @@ class MonitorProtocolError(MonitorError):
     pass
 
 
+class MonitorCmdError(MonitorError):
```

Comment:
Hi @pevogam 
Thanks for pointing out this mistake.
I have one question the `class MonitorCmdError` looks like includes the `class QMPCmdError` https://github.com/avocado-framework/avocado-vt/blob/f618e085b9203e9d2210252bedf1664049b83518/virttest/qemu_monitor.py#L84 by comparing the job what they handle.
Maybe we should find a way to distinguish them or make `QMPCmdError` inherit from this new class.
Or new `class HMPCmdError(MonitorError):` ?

Reply:
Hi @YongxueHong and thanks for reviewing!

> Maybe we should find a way to distinguish them or make QMPCmdError inherit from this new class.

This looks like the easier option without having to create an entire new exception hierarchy just for the human monitor. It also doesn't make sense to create new exception classes if they don't bring any additional utility and distinction in comparison to the old ones so this sounds like the better option for me.

> Or new class HMPCmdError(MonitorError): ?

This would end up branching out an entire HMP exception hierarchy and so far I can't see what we would gain from having identically behaving exceptions for QMP and HMP that are then just separated by name.

Reply:
> Hi @YongxueHong and thanks for reviewing!
> 
> > Maybe we should find a way to distinguish them or make QMPCmdError inherit from this new class.
> 
> This looks like the easier option without having to create an entire new exception hierarchy just for the human monitor. It also doesn't make sense to create new exception classes if they don't bring any additional utility and distinction in comparison to the old ones so this sounds like the better option for me.
> 

Hi @pevogam 
From my considering, how do we give the definition both of `MonitorCmdError` and `QMPCmdError` ?
I mean which scenario should call the `MonitorCmdError`  and which should call `QMPCmdError` ?
Yeah, for the `QMPCmdError`, I can get the purpose from the naming, but for the `MonitorCmdError`, I can not sure, if there are other monitor types involved in the future, like `XXXMonitor` then can we call the `MonitorCmdError` by default?

BTW there are two kinds of monitor inherited from `class Monitor(object)`, those are `class QMPMonitor(Monitor):` and `class HumanMonitor(Monitor):` so I think whether we can also refer to this way to design for the part of CmdError?

Thanks.
> > Or new class HMPCmdError(MonitorError): ?
> 
> This would end up branching out an entire HMP exception hierarchy and so far I can't see what we would gain from having identically behaving exceptions for QMP and HMP that are then just separated by name.

Reply:
I agree with you if separating `QMPCmdError` and `HumanCmdError` makes sense as semantics. At present the exceptions are identical in structure and we are only creating them as labels with no meaningful separation taking place. In short, we would define `QMPCmdError` and `HumanCmdError` classes only to follow the heuristics from separate monitor types, yet the classes themselves are both rather a shared `MonitorCmdError` with no distinction. So I would suggest `QMPCmdError` subclassing the `MonitorCmdError` or being dropped altogether and both monitor classes just using the `MonitorCmdError` type. The rest would be artificial separation of concerns. What do you think?